lib/controllers/run-controller.ts

```diff
@@ -58,6 +73,11 @@ export class RunController extends EventEmitter implements IRunController {
 		const { projectDir, deviceIdentifiers, stopOptions } = data;
 		const liveSyncProcessInfo = this.$liveSyncProcessDataService.getPersistedData(projectDir);
 		if (liveSyncProcessInfo && !liveSyncProcessInfo.isStopped) {
+			if (this.prepareReadyEventHandler) {
```

Comment:
I think this code should be several lines below - when we actually stop the livesync. Currently, it is possible to stop the LiveSync per device, so the handler should remain for other devices.